src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -862,5 +864,11 @@ private static Complex Scale(Complex value, double factor)
         {
             return new Complex((double)value, 0.0);
         }
+
+        public static bool IsFinite(Complex value) => !Infinity.Equals(value);
+
+        public static bool IsNaN(Complex value) => NaN.Equals(value);
```

Comment:
This can probably be implemented using `!IsInfinity(value) && !IsFinite(value)`. The check basically needs to return true when either real or imaginary is NaN and neither is Infinite.